Refactors Markings, Adds Tail (some body and head) markings, Adds Alt heads, Secondary (F)Hair Themes, Darkens Vulp/Taj + Unathi horns/frills, Sprite adjustments and more...

KasparoVy opened the pull request:
**_CREDITS:_**
Tail marking idea and Vox tail marking sprites by @Fullofskittles.
Most of the new body and head marking sprites by @Fullofskittles.
5 new Tajaran hair sprites from **Polaris(?)**.
Sharp Unathi snout inspired by **TG**.

## IMPORTANT NOTICE:
**Vulpkanin and Tajara body sprites have been darkened! Characters will have to adjust their body colours if they want to look the same.**
- _Add 27 to the R, G and B values if Vulpkanin._
- _Add 15 to the R, G and B values if Tajara._

**Tiger body markings have been darkened! Characters will have to adjust their body marking colours if they want to look the same.**
- _Add 14 to the R, G and B values if body/head marking has 'tiger' in it._

**Vulpkanin facial fur patterns have been darkened! Characters will have to adjust their head accessory/facial hair colours if they want to look the same.**
- _Add 23 to the R, G and B values of your facial hair/head accessories._

**Unathi frills and horns have been darkened! Characters will have to adjust their facial hair colours if they want to look the same.**
- _Add 27 to the R, G and B values of your frills. You're going to have to do the webbing colour from scratch._
- _Add 21 to the R, G and B values of your horns._

**Tajaran head/facial hair has been darkened! Characters will have to adjust their head/facial hair colours if they want to look the same.**
- _Add 28 to the R, G and B values of your head/facial hair._

**Vulpkanin head hair has been darkened! Characters will have to adjust their head hair colours if they want to look the same.**
- _If Adhara, add 12 to R, G and B values for hair colour._
- _If Anita, add 10 to R, G and B values for hair colour._
- _If Apollo, add 6 to R, G and B values for hair colour._
- _If Belle, add 6 to R, G and B values for hair colour._
- _If Bun, add 5 to R, G and B values for hair colour._
- _If Curl, add 15 to R, G and B values for hair colour._
- _If Hawk, don't do anything._
- _If Jagged, add 10 to R, G and B values for hair colour._
- _If Kajam, add 16 to R, G and B values for hair colour._
- _If Keid, add 27 to R, G and B values for hair colour._
- _If Kleeia, add 14 to R, G and B values for hair colour._
- _If Mizar, add 14 to R, G and B values for hair colour._
- _If Short, add 7 to R, G and B values for hair colour._
- _If Spike, add 6 to R, G and B values for hair colour._

## Features, fixes and changes:
:cl:
tweak: Refactors markings. Now split into head, body and tail markings.
tweak: Refactors morph again. Changes made while morphing are now reflected instantly on your sprite.
tweak: Darkens Vulpkanin and Tajara for improved colour fidelity. Species base colour changed to compensate. Add 27 to R, G and B values if Vulp, 15 if Taj to keep the same colour on your character.
tweak: Darkens Vulpkanin facial fur patterns for improved colour fidelity. Add 23 to the R, G and B values of your facial hair/head accessory to keep the same colour on your character.
tweak: Darkens Unathi frills and horns. Add 27 to the R, G and B values of your frills and 21 for horns to keep the same colour on your character.
tweak: Vulpkanin tails are now split-rendered. Tails overlap hair when facing north, but are overlapped by hair when facing all other directions. No other differences, only affects the fluffy husky tail.
tweak: The heads of tiger markings have been isolated and can now be coloured independently.
tweak: Hair can now cover facial hair, and glasses are no longer covered by facial hair and head accessories. Means that two-piece Vulpkanin facial fur patterns won't almost entirely cover whatever glasses they're wearing.
tweak: Tajaran head and facial hair have been darkened. Add 28 to the R, G and B values of your Tajara head/facial hair to keep the same colour on your character.
tweak: Vulpkanin head hair has been darkened. Refer to PR notes for darkening factors.
fix: Naked Humans and Skrell will no longer have exposed flesh poking out their wrists when facing east or west. Male Humans now no longer lack a butt pixel when facing east or west.
fix: Mutated limbs now render correctly as soon as they're mutated.
fix: Species with TAIL_OVERLAPPED will no longer wag tail at max speed when facing north.
fix: Species with TAIL_OVERLAPPED and body accessories won't crash the game when they wag.
fix: Tiger Head and Face markings adjusted so they don't look ugly on certain species' ears.
fix: Fixes issue where Unathi dorsal stripe didn't render.
fix: Tajaran ears are no longer in their hair sprites, but on their actual heads.
fix: The adminbus body accessory now overrides the TAIL_OVERLAPPED flag and will now be rendered correctly on a body regardless of species bodyflags.
rscadd: Adds some head markings for Tajara and Vulpkanin.
rscadd: Adds body markings for Tajara, Vulpkanin and Unathi.
rscadd: Adds tail markings for Vox and Vulpkanin.
rscadd: Adds 5 Tajaran hairstyles.
rscadd: Adds a hairstyle for Humans, Unathi and Vulpkanin.
rscadd: Adds secondary hair (head and facial) themes. Skrell can now colour their tentacle cloths independently, the beads in the Tajaran/Hippie braid hairstyles can be coloured and the same with the webbing in Unathi frills. Can be changed at nanomirrors, via morph and CMA.
rscadd: Adds alternate heads. Unathi can now choose a head with a sharp snout. Can be changed via morph and CMA.
/:cl:

## Required SQL Changes:
```
ALTER TABLE `characters` CHANGE COLUMN `marking_style_name` `marking_styles` mediumtext NOT NULL DEFAULT 'head=None&body=None&tail=None';
ALTER TABLE `characters` DROP COLUMN `markings_blue`, DROP COLUMN `markings_green`;
ALTER TABLE `characters` CHANGE COLUMN `markings_red` `marking_colours` mediumtext NOT NULL DEFAULT 'head=%23000000&body=%23000000&tail=%23000000';
UPDATE `characters` SET `hair_style_name` = "Bald" WHERE `hair_style_name` = "Tajaran Ears";
ALTER TABLE `characters` ADD COLUMN `alt_head_name` varchar(45) NOT NULL AFTER head_accessory_style_name;
UPDATE `characters` SET `hair_style_name` = "Bald" WHERE `hair_style_name` = "None";
ALTER TABLE `characters` ADD COLUMN `secondary_facial_red` smallint(4) NOT NULL AFTER facial_blue;
ALTER TABLE `characters` ADD COLUMN `secondary_facial_green` smallint(4) NOT NULL AFTER secondary_facial_red;
ALTER TABLE `characters` ADD COLUMN `secondary_facial_blue` smallint(4) NOT NULL AFTER secondary_facial_green;
ALTER TABLE `characters` ADD COLUMN `secondary_hair_red` smallint(4) NOT NULL AFTER hair_blue;
ALTER TABLE `characters` ADD COLUMN `secondary_hair_green` smallint(4) NOT NULL AFTER secondary_hair_red;
ALTER TABLE `characters` ADD COLUMN `secondary_hair_blue` smallint(4) NOT NULL AFTER secondary_hair_green;
```

## More pictures coming.
_Vulpkanin Darkening_
![vulpdarkening](https://cloud.githubusercontent.com/assets/12377767/16969884/2e031cdc-4de6-11e6-869b-df661182a518.png)

_The reason why Vulpkanin tails are now split-rendered._
![vulptailsplit](https://cloud.githubusercontent.com/assets/12377767/16969893/3c3ee95c-4de6-11e6-9e7e-ce6a6018281e.png)

_Vulpkanin chargen._
![markingsvulp](https://cloud.githubusercontent.com/assets/12377767/16974218/6638df66-4e08-11e6-8fa9-60bc9209d3aa.PNG)

_Unathi chargen. The 'colour # 2' options only pop up if the selected hair/facial hair style has secondary themes and can be coloured._
![unathichargen](https://cloud.githubusercontent.com/assets/12377767/16974246/adb6a242-4e08-11e6-846b-664a5694f42f.PNG)

_Unathi in-game feat. hair with secondary theme (beads) and facial hair with secondary theme (webbing)._
![unathiingameid](https://cloud.githubusercontent.com/assets/12377767/16974501/31c36fdc-4e0b-11e6-8bc6-6c0c7682f1cf.png)

_Vulpkanin in-game feat. hair with secondary theme (beads), chest crest and tail markings._
![vulpingame](https://cloud.githubusercontent.com/assets/12377767/16974560/b701e00c-4e0b-11e6-8e1d-f857b0a0d849.png)

_Vulpkanin Tail-marking anims working as intended._
![vulptailmarkingwag](https://cloud.githubusercontent.com/assets/12377767/16974905/87874606-4e0f-11e6-8767-836bef7b5dde.gif)

_Vox Tail-marking anims working as intended. No more top-speed wag when facing north._
![voxtailmarkingsstatic](https://cloud.githubusercontent.com/assets/12377767/16975014/847151ae-4e10-11e6-889d-5607fcf68b3e.png)![voxtailmarkingwag](https://cloud.githubusercontent.com/assets/12377767/16975015/8471cef4-4e10-11e6-8fb7-1c7aeb4e1e08.gif)![voxtailmarkingswag](https://cloud.githubusercontent.com/assets/12377767/16975016/8472234a-4e10-11e6-9bff-5757bde5f9ac.gif)

_Seeing that we're able to change tail markings in the admin CMA panel._
![cmatailmarkings](https://cloud.githubusercontent.com/assets/12377767/16975123/60870fb2-4e11-11e6-8dc1-5f55a4a36d55.PNG)

_Seeing that we're able to change LOTS of stuff in the admin CMA panel. Keep in mind that you can change secondary hair and secondary facial hair colours at a SalonPro Nanomirror(tm) and that you can change all this via Morph as well._
![unathiadmincma](https://cloud.githubusercontent.com/assets/12377767/16975128/6f8d6178-4e11-11e6-8c62-3334ea68e9ef.PNG)

_Skrell can now independently colour their tentacle cloths. Was going to be part of another PR but the idea fit in much better here._
![skrelltentaclesclothcolour](https://cloud.githubusercontent.com/assets/12377767/16976020/c1797762-4e19-11e6-86fe-e8f151b0eb5e.png)

_Darkened Tajara as well in the same manner as Vulpkanin._
![tajdarkening](https://cloud.githubusercontent.com/assets/12377767/16979162/9d74b1ba-4e2d-11e6-87ec-a1b887f7279b.png)

_Tattoos using Humans as an example species._
![tatsonhumans](https://cloud.githubusercontent.com/assets/12377767/17078758/51c898aa-50c9-11e6-900c-d7dc0569cc29.png)

_Hippie braid hairstyle (including the OG on Tajara. Also showcases some of the other new cosmetic features.)_
![hippiebraid](https://cloud.githubusercontent.com/assets/12377767/17079648/cef33d1a-50e4-11e6-92e9-fcfbb08bcfb1.png)

_The new Tajaran hair styles from Polaris._
_Top row: L - Curly, Mid - Ladies' Retro, R - Victory Curls_
![tajaranhair](https://cloud.githubusercontent.com/assets/12377767/17079649/e1d6113c-50e4-11e6-9041-9d0b6c2f98a8.png)
_Bottom row: L - Bob, Mid - Finger Curls_

_Foxes! Showcasing more of the markings._
![foxes](https://cloud.githubusercontent.com/assets/12377767/17081531/50ffc6fc-512b-11e6-8ca6-41c57859ea65.png)
![foxwag1](https://cloud.githubusercontent.com/assets/12377767/17081532/53e5fb16-512b-11e6-8e68-f3afee4bab69.gif)![foxwag2](https://cloud.githubusercontent.com/assets/12377767/17081533/53f2cb02-512b-11e6-824d-3d17d4c18a90.gif)

_The rest of the Vulpkanin tail markings._
![othertailmarkings](https://cloud.githubusercontent.com/assets/12377767/17081537/703d732a-512b-11e6-9037-8443a5f22e1f.png)
![silverfoxwag1](https://cloud.githubusercontent.com/assets/12377767/17081538/75890f4c-512b-11e6-9258-bbc712e229fb.gif)![silverfoxwag3](https://cloud.githubusercontent.com/assets/12377767/17081539/759a007c-512b-11e6-95ce-a512215e17fc.gif)![silverfoxwag2](https://cloud.githubusercontent.com/assets/12377767/17081540/75a02196-512b-11e6-844e-1a21387025ee.gif)

_Glasses are no longer covered by facial hair and head accessories. You can now see glasses on Vulpkanin._
![glassesfix](https://cloud.githubusercontent.com/assets/12377767/17238187/94607216-5527-11e6-9fb9-f81ce14318d5.png)

KasparoVy commented:
More pictures to be added

KasparoVy commented on code/_onclick/hud/robot.dm:
```diff
@@ -1,241 +1,241 @@
-/obj/screen/robot
```
I have no idea why this happened. It deletes and readds everything with no differences, and I never even opened this file.

Fox-McCloud replied:
something went wrong

Fox-McCloud replied:
This will still need to be addressed; I recommended deleting this file, outright, then downloading it directly off Para's git website and putting it back in to see if it resolves this.

KasparoVy replied:
Alright I'll get on that now, I did it with raw paste and that didn't work so I'll try this method too

Fox-McCloud commented:
How is this going to impact those who already have the exact color we like for our characters? I took quite a while tweaking my character to be the proper color--will I have to re-tweak it, again, to get it back to that?

KasparoVy commented:
For body colour, it's now 25 lum darker for vulpkanin-- so if you add 27 to each RGB value, it should get you back to your colour. I think it's the same number for Tajara.

Aurorablade commented:
I demand taja tail jewelery

KasparoVy commented:
Absolutely possible.

FlattestGuitar commented:
Now you're crapping on Jay's parade. He and his snowflakey tail tattoo will no longer be so speshul.

Fox-McCloud commented:
there totally needs to be a tail tip marking for the default Vulp tail

totally not biased here or anything :wink:

KasparoVy commented:
@Fox-McCloud Yup, I've got it set up. Only reason why I've got that list of commands, haha. I have it in order of what I entered, as well. Also, way ahead of you. There's already a tail tip marking for the default Vulp tail in this PR, one of the things I need a picture for. No guarantees on quality, but it fits the bill.

Aurorablade commented:
I SAID TAIL JEWLERY NOT TATOOS..AS LONG AS WE DO NOT PUT IN TAIL TATOOS WE ARE GOOD...

KasparoVy commented:
I know what you meant and I've got some ideas, no worries.

Jaywings commented:
M-more Tajaran stuff...my dream come true.

Aurorablade commented:
*pulls jay into the cat box fort..*

Lady-Luck commented:
Can unathi get more hair styles? Pretty much every unathi woman goes around with beehive.

Aurorablade commented:
as a unathi...plz no no more unathi hair....

Lady-Luck commented:
>as a unathi...plz no no more unathi hair....

Uh... why? You got all the women using the same hairstyle.

Aurorablade commented:
yeah for the few of them that do you hair..I don't..we have plenty of horn and frill styles to choose from.

Lady-Luck commented:
People who want hair on their unathi will continue wanting hair on their unathi, so more styles would be good so all the people who want hair on their unathi have more than one feminine hairstyle to choose from.

Aurorablade commented:
....we could use more horn and frill styles though..i mean i LIKE my ramhorns..buth..

KasparoVy commented:
I get what you mean, but if you're referring to porting more Unathi hair styles from a place like Polaris, Bay or Aurora, we don't have any less hair styles Unathi can use than them. I think a good idea would be to make a thread, see what hair styles people'd like most for Unathi, then work from there.

Lady-Luck commented:
>I get what you mean, but if you're referring to porting more Unathi hair styles from a place like Polaris, Bay or Aurora, we don't have any less hair styles Unathi can use than them. I think a good idea would be to make a thread, see what hair styles people'd like most for Unathi, then work from there.

Hmm. I'll ask some spriters how the hair system works, maybe I could just make some minor adjustments to get some basic human hairstyles converted. If I did this, would you add them?

KasparoVy commented:
Absolutely, yeah. I've already added one (the hippie braid) in a similar manner, so I've certainly no problem doing it for others

Fox-McCloud commented:
@KasparoVy 

Get your butt on skype, want to run a few things by you. :stuck_out_tongue:

Aurorablade commented:
No. Unathi. Hair. Please

Lady-Luck commented:
No one's forcing you to use them.

Aurorablade commented:
Yeah  but I would be forced to look at it like a hot pink fur color. I could accept it if the hair looked like quills or spines

KasparoVy commented:
@Aurorablade Hair on Unathi = wig, spines = legit? It's certainly an idea to hold onto, I think it'd make for some interesting sprites.

Lady-Luck commented:
You're already looking at hair Aurorablade, unathi can have hair. It just gives more options for the hair-users.

Aurorablade commented:
Well I disagree with hair on unathi

Spacemanspark commented:
I disagree with anything that is bright neon pink.
The PR itself looks nice.

KasparoVy commented:
Computer crashed while I was working and totally corrupted my entire GIT history, diagram and all that. Taking a bit of time to wipe away the debris and clone my fork again.

KasparoVy commented:
More pictures added. Pics of the Tajaran hairstyles are next and then I'll move on to markings

KasparoVy commented:
Tajaran hair style pictures added.

Fox-McCloud commented:
@KasparoVy 

Have pics of the default Vulp tail with tail tip marking?

KasparoVy commented:
I'll whip one up right now, sorry for the wait.

KasparoVy commented:
![foxes](https://cloud.githubusercontent.com/assets/12377767/17080452/438c9e3c-50fe-11e6-9cdc-4103cfbaf9dd.png)

![foxwag2](https://cloud.githubusercontent.com/assets/12377767/17080453/54f92c30-50fe-11e6-9cab-538d110bd5ed.gif)![foxwag1](https://cloud.githubusercontent.com/assets/12377767/17080454/54fae232-50fe-11e6-869b-a4a46baec271.gif)

KasparoVy commented:
@Fox-McCloud
![foxes](https://cloud.githubusercontent.com/assets/12377767/17080452/438c9e3c-50fe-11e6-9cdc-4103cfbaf9dd.png)

![foxwag2](https://cloud.githubusercontent.com/assets/12377767/17080453/54f92c30-50fe-11e6-9cab-538d110bd5ed.gif)![foxwag1](https://cloud.githubusercontent.com/assets/12377767/17080454/54fae232-50fe-11e6-869b-a4a46baec271.gif)

Fox-McCloud commented:
Thanks!

tigercat2000 commented on code/modules/mob/living/carbon/human/update_icons.dm:
```diff
@@ -299,7 +299,7 @@ var/global/list/damage_icon_parts = list()
 	//END CACHED ICON GENERATION.
 	stand_icon.Blend(base_icon,ICON_OVERLAY)
-	if(src.species.bodyflags & TAIL_OVERLAPPED) // If the user's species is flagged to have a tail that needs to be overlapped by limbs...
+	if(((body_accessory && istype(body_accessory, /datum/body_accessory/tail)) || !body_accessory) && species.bodyflags & TAIL_OVERLAPPED) // If the user's species is flagged to have a tail that needs to be overlapped by limbs... (having a non-tail body accessory like the snake body will override this)
```
um
shouldn't this just be
```
if((!body_accessory || istype(body_accessory, /datum/body_accessory/tail)) && species.bodyflags & TAIL_OVERLAPPED)
```

KasparoVy replied:
Is the istype okay without a check to see if body_accessory exists beforehand? Haven't had to deal with that before and it worked just fine the way it currently is. I'll test it and change if it this other way works, thanks for pointing it out

Fox-McCloud commented:
Just let us know when this is ready for review.

KasparoVy commented:
Ready for review. All the sprites are in order, all the lists are populated correctly, CMA and morph seem to be doing their stuff just fine and everyone who should be accessing the new features can so I guess that's all I've got unless I missed a spot. I'll keep looking, myself.

KasparoVy commented:
Going to have to compensate for the parallax merge/schema fix, commit coming shortly

KasparoVy commented:
Thought this was going to conflict with parallax, quite relieved it didn't. Updated and tested just to make sure, and all seems well.

KasparoVy commented:
Now that's REALLY all I've got. That last bit was something that I noticed was wrong IG and back while I was doing the screenshots.

KasparoVy commented:
Vulp hairstyles
---------------------------
Adhara - Darkened by 12
Anita - Darkened by 10
Apollo - Darkened by 6
Belle - Darkened by 6
Bun - Darkened by 5
Curl - Darkened by 15
Hawk - Not darkened.
Jagged - Darkened by 10
Kajam - Darkened by 16
Keid - Darkened by 27
Kleeia - Darkened by 14
Mizar - Darkened by 14
Short - Darkened by 7
Spike - Darkened by 6

KasparoVy commented:
That's it, I SWEAR. Missed that while I was darkening the bodies. I'm certain now that I've forgotten nothing else. :sweat_smile: 

**Vulpkanin head hair has been darkened! Characters will have to adjust their head hair colours if they want to look the same.**
- _If Adhara, add 12 to R, G and B values for hair colour._
- _If Anita, add 10 to R, G and B values for hair colour._
- _If Apollo, add 6 to R, G and B values for hair colour._
- _If Belle, add 6 to R, G and B values for hair colour._
- _If Bun, add 5 to R, G and B values for hair colour._
- _If Cur, add 15 to R, G and B values for hair colour._
- _If Hawk, don't do anything._
- _If Jagged, add 10 to R, G and B values for hair colour._
- _If Kajam, add 16 to R, G and B values for hair colour._
- _If Keid, add 27 to R, G and B values for hair colour._
- _If Kleeia, add 14 to R, G and B values for hair colour._
- _If Mizar, add 14 to R, G and B values for hair colour._
- _If Short, add 7 to R, G and B values for hair colour._
- _If Spike, add 6 to R, G and B values for hair colour._

Allfd commented:
@KasparoVy 
Really looking forward to the flexibility this will bring.

taukausanake commented:
> fix: Tajaran ears are no longer in their hair sprites, but on their actual heads.

Whoa I missed this part. I tried doing this myself one time and I didn't like how it looked. It works on Vulpkanin but for Tajaran it looks kind of bad. I don't know exactly why but I think it has to do with Tajaran being more short-hair than Vulpkanin are. I mean most of Taj hair styles are small fuzzies that don't take the full of their head while Vulp hairs are a lot longer. I'd ask to look into this more

KasparoVy commented:
The hair sprites have the holes cut out for the Tajara ears so there shouldn't be any issue with hair hiding their ears if that's what your wondering. If you refer to the screenshots featuring Tajara you'll see what I mean: It looks unchanged except the ears are now attached to the body and inherit body colour rather than being a part of a hair style and inheriting hair colour and requiring a separate head accessory to be overlain ontop of it configured with, again, the character's body colour in order to achieve the desired effect *gasp*

**tl;dr - See cat screenshots**

KasparoVy commented:
The hair sprites have the holes cut out for the Tajara ears so there shouldn't be any issue with hair hiding their ears if that's what your wondering. If you refer to the screenshots featuring Tajara you'll see what I mean: It looks unchanged except the ears are now attached to the body and inherit body colour rather than being a part of a hair style and inheriting hair colour and requiring a separate head accessory to be overlain ontop of it configured with, again, the character's body colour in order to achieve the desired effect *gasp*

**tl;dr - See cat screenshots**

@Fox-McCloud Your suggestion to replace the file worked, thanks! I navigated to https://raw.githubusercontent.com/ParadiseSS13/Paradise/master/code/_onclick/hud/robot.dm and saved the webpage as a .dm file, deleted the one in my local branch and dropped in the downloaded version and it sorted it out. Confirmed down one file change in the 'files' tab :+1:

taukausanake commented:
> except the ears are now attached to the body and inherit body colour rather than being a part of a hair style and inheriting hair colour

This is what I mean. On Vulps this looks fine, different hair and ear colors. With Tajs it didn't look nice to me. Maybe it doesn't matter though but that's just what I know

KasparoVy commented:
@Taukausanake Then there's still head accessories that let you colour the ears specifically in 3 different ways so you can colour the ears whatever way you like

taukausanake commented:
Huh. Yeah I guess there was that. Okay, forgot or didn't see its usage back then

KasparoVy commented:
No worries I'm glad you asked, it's good to clear that sort of stuff up just in case others were wondering too

KasparoVy commented:
There, much better. Changed the 'required SQL changes' section as necessary

TheDZD commented on code/modules/mob/living/carbon/human/appearance.dm:
```diff
@@ -72,44 +54,59 @@
 /mob/living/carbon/human/proc/change_body_accessory(var/body_accessory_style)
 	var/found
-	if(!body_accessory_style)
+	if(!body_accessory_style || (src.body_accessory && src.body_accessory.name == body_accessory_style))
```
`src.` isn't needed, and shouldn't ever be, assuming coders don't name their proc arguments the same as their variables.
